lib/cppcheck.cpp

```diff
@@ -1012,8 +1016,8 @@ unsigned int CppCheck::checkFile(const FileWithDetails& file, const std::string
         reportErr(errmsg);
     }
 
-    if (!mSettings.buildDir.empty()) {
-        mAnalyzerInformation.close();
+    if (mAnalyzerInformation) {
```

Comment:
Nit: I guess we don't need the condition. We can just reset directly and let it check internally..?

Reply:
Makes sense - but I intentionally kept it intact for now. The future refactorings might get rid of it.